script/verification/HoloceneSystemConfigUpgrade.s.sol

```diff
@@ -0,0 +1,279 @@
+    function checkScalar() internal view {
+        // Check that basefeeScalar and blobbasefeeScalar are correct by re-encoding them and comparing to the new scalar value.
+        uint256 reencodedScalar =
+            (uint256(0x01) << 248) | (uint256(sysCfg.blobbasefeeScalar()) << 32) | sysCfg.basefeeScalar();
+        console.log(
+            "checking baseFeeScalar and blobbaseFeeScalar reencoded to ",
+            LibString.toString(sysCfg.basefeeScalar()),
+            LibString.toString(sysCfg.blobbasefeeScalar()),
+            LibString.toString(reencodedScalar)
+        );
+        uint256 newScalar = sysCfg.scalar();
+        require(newScalar == reencodedScalar, "scalar-105");
+
+        // Next, we check that the scalar itself was migrated properly
+        uint8 previousScalarEncodingVersion = uint8(previousScalar >> 248);
+        uint8 newScalarEncodingVersion = uint8(newScalar >> 248);
+
+        if (
+            // If the scalar version (i.e. the most significant bit of the scalar)
+            // is 1, we expect it to be unchanged during the upgrade.
+            previousScalarEncodingVersion == 1
+        ) {
+            require(reencodedScalar == previousScalar, "scalar-100 scalar mismatch");
+        } else {
+            // Otherwise, if the scalar version is 0,
+            // the blobbasefeeScalar is implicitly 0 and
+            // the upgrade will migrate the scalar version to 1 and preserve
+            // everything else.
+            // See https://specs.optimism.io/protocol/system-config.html?highlight=ecotone%20scalar#ecotone-scalar-overhead-uint256uint256-change
+            require(previousScalarEncodingVersion == 0, "scalar-101 previous scalar version != 0 or 1");
+            require(newScalarEncodingVersion == 1, "scalar-102 reenconded scalar version != 1");
+            require(sysCfg.blobbasefeeScalar() == uint32(0), "scalar-103 blobbasefeeScalar !=0");
+
+            // The scalars should match if we add the new scalar version byte to the previous scalar.
+            require(reencodedScalar == previousScalar + (uint256(0x01) << 248), "scalar-104 scalar mismatch");
+        }
+    }
+
+    // Checks the disputeGameFactory address is set correctly after the upgrade
+    function checkDGF() internal view {
+        require(sysCfg.disputeGameFactory() == targetDGF, "scalar-106");
+    }
+
+    // Checks gasPayingToken and its decimals are set correctly after the upgrade
+    function checkGasPayingToken() internal view {
+        (address t, uint8 d) = sysCfg.gasPayingToken();
+        require(t == targetGasPayingToken, "scalar-107");
+        require(d == targetDecimals, "scalar-108");
+    }
+
+    // Checks the remaining storage variables are unchanged after the upgrade
+    function checkBaseSysCfgVars() internal view {
+        // Check remaining storage variables didn't change
+        require(keccak256(abi.encode(getBaseSysCfgVars())) == keccak256(abi.encode(previous)), "system-config-100");
```

Comment:
```suggestion
        require(getBaseSysCfgVars().eq(previous), "system-config-100");
```

Reply:
I don't think this one will work because I'm not ` using LibString for BaseSysCfgVars` and I don't think I can either.

Reply:
ah lol, I realize this is not a string comparison, sorry!